pkg/service/presentation/storage/bolt.go

```diff
@@ -20,6 +20,63 @@ type BoltPresentationStorage struct {
 	db *storage.BoltDB
 }
 
+type opUpdater struct {
+	storage.UpdaterWithMap
+}
+
+func (u opUpdater) SetUpdatedResponse(first []byte) {
+	u.UpdaterWithMap.Values["response"] = first
+}
+
+func (u opUpdater) Validate(v []byte) error {
+	var op opstorage.StoredOperation
+	if err := json.Unmarshal(v, &op); err != nil {
+		return errors.Wrap(err, "unmarshalling operation")
+	}
+
+	if op.Done {
+		return errors.New("operation already marked as done")
+	}
+
+	return nil
+}
+
+var _ storage.ResponseSettingUpdater = (*opUpdater)(nil)
```

Comment:
never quite got the point of these can you help me understand?

Reply:
do you get a compile err without it?

Reply:
thanks - makes sense. should probably get this pattern in use in a few other places for safety